src/Interpreters/IKeyValueEntity.h

```diff
@@ -10,32 +12,27 @@ namespace DB
 class IKeyValueEntity
 {
 public:
-    IKeyValueEntity() = default;
     virtual ~IKeyValueEntity() = default;
 
-    /// Get primary key name that supports key-value requests.
-    /// Primary key can constist of multiple columns.
+    /** Get primary key name that supports key-value requests.
+      * Primary key can constist of multiple columns.
+      */
     virtual Names getPrimaryKey() const = 0;
 
     /*
-     * Get data from storage directly by keys.
+     * Get columns from storage using keys columns.
      *
-     * @param keys - keys to get data for. Key can be compound and represented by several columns.
-     * @param out_null_map - output parameter indicating which keys were not found.
-     * @param required_columns - if we don't need all attributes, implementation can use it to benefit from reading a subset of them.
+     * @param attribute_names - if we don't need all attributes, implementation can use it to benefit from reading a subset of them.
+     * If attribute_names are empty, return all attributes.
+     * @param key_columns - keys to get data for. Key can be compound and represented by several columns.
+     * @param found_keys_map - output parameter indicating which keys were not found. 1 - key found. 0 - key is not found.
      *
      * @return - chunk of data corresponding for keys.
      *   Number of rows in chunk is equal to size of columns in keys.
      *   If the key was not found row would have a default value.
      */
-    virtual Chunk getByKeys(const ColumnsWithTypeAndName & keys, PaddedPODArray<UInt8> & out_null_map, const Names & required_columns) const = 0;
+    virtual Block getColumns(const Names & attribute_names, const ColumnsWithTypeAndName & key_columns, PaddedPODArray<UInt8> & found_keys_map) const = 0;
```

Comment:
@vdimir I changed interface here to return Block instead of Chunk in single method.
Main motivation is that current usages are get chunks then get their sample block.
Key value entity interface does not provide any lock mechanism. And it could be possible that you use `getColumns` get some result chunk, and in the same time storage was modified and during `getSampleBlock` call you get completely different Block.